protofsm/state_machine.go

```diff
@@ -721,3 +705,11 @@ func (s *StateMachine[Event, Env]) driveMachine(ctx context.Context) {
 		}
 	}
 }
+
+// decorateCtx adds the FSM name to the context as a logging attribute.
+func (s *StateMachine[Event, Env]) decorateCtx(
+	ctx context.Context) context.Context {
+
+	return btclog.WithCtx(ctx,
+		btclog.Fmt("fsm_name", s.cfg.Env.Name())) //nolint:govet
```

Comment:
Is there anyway we can keep the old prefix nature? At least when scanning, IMO it's useful for it to be at the very start as depending on the size of your editor/viewport, the end of the line may require moving the cursor to actually view.

Reply:
ok i've added a temporary commit at the end that does it this way. Let me know if you think this is ok and i can squash in the change. 

The logs now look like this:

```
2024-12-10 13:20:44.552 [DBG]: FSM(test) Starting state machine
2024-12-10 13:20:44.552 [DBG]: FSM(test) Sending event event=(*protofsm.emitInternal)(0x1045f1240)({
})

2024-12-10 13:20:44.552 [DBG]: FSM(test) Applying new event event=(*protofsm.emitInternal)(0x1045f1240)({
})

2024-12-10 13:20:44.552 [DBG]: FSM(test) Processing event event=(*protofsm.emitInternal)(0x1045f1240)({
})

2024-12-10 13:20:44.552 [DBG]: FSM(test) Adding new internal event to queue event !BADKEY=(*protofsm.goToFin)(0x1045f1240)({
})

2024-12-10 13:20:44.552 [INF]: FSM(test) State transition from_state=*protofsm.dummyStateStart to_state=*protofsm.dummyStateStart
2024-12-10 13:20:44.552 [DBG]: FSM(test) Processing event event=(*protofsm.goToFin)(0x1045f1240)({
})

2024-12-10 13:20:44.552 [INF]: FSM(test) State transition from_state=*protofsm.dummyStateStart to_state=*protofsm.dummyStateFin
```

Reply:
squashed in the temp commit